manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -271,10 +284,16 @@ func (tr *TaskReaper) tick() {
 				if int64(len(historicTasks)) <= taskHistory {
 					break
 				}
-			}
 
-			if runningTasks <= 1 {
-				delete(tr.dirty, dirty)
+				// The only case when we keep the slot dirty at the end of tick()
+				// is when there are more than one running tasks present
+				// for a given slot.
+				// In that case, we keep the slot dirty to allow it to be
+				// cleaned when tick() is called next and one or more of
+				// the tasks in that slot have stopped running.
+				if runningTasks <= 1 {
+					delete(tr.dirty, dirty)
+				}
```

Comment:
the indentation level changes here... this statement now executes for every task in historicTasks. Is this intended behavior?

Reply:
oversight on my part. Fixed!